great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1293,3 +1379,70 @@ def get_batch_data_and_markers(
             )
 
         return batch_data, batch_markers
+
+    @contextmanager
+    def get_connection(self) -> sqlalchemy.Connection:
+        """Get a connection for executing queries.
+
+        Some databases sqlite/mssql temp tables only persist within a connection,
+        so we need to keep the connection alive by keeping a reference to it.
+        Even though we use a single connection pool for dialects that need a single persisted connection
+        (e.g. for accessing temporary tables), if we don't keep a reference
+        then we get errors like sqlite3.ProgrammingError: Cannot operate on a closed database.
+
+        Returns:
+            Sqlalchemy connection
+        """
+        if self.dialect_name in _PERSISTED_CONNECTION_DIALECTS:
+            try:
+                if not self._connection:
+                    self._connection = self.engine.connect()
+                yield self._connection
```

Comment:
When/where does the connection get closed?

Reply:
Good question! I think we can make future improvements here to close the connection deliberately, but to keep the PR scope here more limited (to enabling compatibility with sqlalchemy & pandas 2.0+) we are continuing with the existing method of the connection closing during garbage collection (for connections that are persisted i.e. `sqlite` and `mssql`). 
For future improvements, I think closing the connection during the checkpoint validation workflow is more straightforward than the interactive `validator` workflow. There are also different types of temporary tables in mssql that we may want to investigate that may be less ephemeral. Happy to discuss further!